core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/constraints/EnvelopePartConstraintType.java

```diff
@@ -3,4 +3,5 @@
 public enum EnvelopePartConstraintType {
     FLOOR,
     CEILING,
+    MAINTAIN_SPEED,
```

Comment:
What does `MAINTAIN_SPEED` mean in this context? Is it a constraint that "fails" if the envelope differs in either direction? If that's the case, I don't think the name is very clear about it.

A comment would help clarify this either way.

Reply:
It means the constraint is to maintain the given speed. It indeed fails if the speed differs from the given speed.

I'm open to rename it, i'm just not very inspired right now.